capa/render/default.py

```diff
@@ -82,26 +84,68 @@ def render_capabilities(doc: rd.ResultDocument, ostream: StringIO):
         | ...                                                   | ...                                             |
         +-------------------------------------------------------+-------------------------------------------------+
     """
+
+    def load_rules_prevalence(filename: str) -> dict:
+        try:
+            with Path(filename).open("rb") as pickle_file:
+                return pickle.load(pickle_file)
+        except FileNotFoundError:
+            raise FileNotFoundError(f"File '{filename}' not found.")
+        except Exception as e:
+            raise RuntimeError(f"An error occurred while loading '{filename}': {e}")
+
     subrule_matches = find_subrule_matches(doc)
+    rules_prevalence = load_rules_prevalence("./assets/rules_prevalence.pickle")
 
-    rows = []
+    # seperate rules based on their prevalence
+    common = []
+    rare = []
     for rule in rutils.capability_rules(doc):
         if rule.meta.name in subrule_matches:
-            # rules that are also matched by other rules should not get rendered by default.
-            # this cuts down on the amount of output while giving approx the same detail.
-            # see #224
             continue
 
         count = len(rule.matches)
-        if count == 1:
-            capability = rutils.bold(rule.meta.name)
+        matches = f"({count} matches)" if count > 1 else ""
+
+        rule_prevalence = float(rules_prevalence.get(rule.meta.name + "\n", 0) / 593)
+        if rule_prevalence < 0:
+            raise ValueError("Match probability cannot be negative")
+
+        prevalences = [rutils.bold("rare"), rutils.bold("common"), "unknown"]
+
+        if rule_prevalence >= 0.05 or rule_prevalence == 0:
```

Comment:
we can set the value here to filter for `rare` and `common` rules. `0` is default value for rules with no prevalence, hence `unknown`